### DISCUSSION TITLE
WSLg is not installed when enabled in "Turn Windows features on or off" window

### DISCUSSION THREAD

**Discussion by IcyCod** [microsoft/wslg#537] [Q&A]
~~Hi everyone, I have an issue which GUI apps do not launch for the reason "cannot open display" (or something similar).~~

~~I followed [this](https://github.com/microsoft/wslg/wiki/Diagnosing-%22cannot-open-display%22-type-issues-with-WSLg) and the problem still occurs. At the start, $DISPLAY was not set, and I set it to ":0". The directory "/mnt/wlsg" is also missing, with only 3 subdirectories under "/mnt" ("c", "g", and "wsl").~~

I enabled the feature in "Turn Windows features on or off" yesterday and started using it. However, apps were not able to launch in GUI mode, and seemed wslg was missing. Using the command `wsl --update` did not update too, and just told that automatic update is off which toggled by "Receive updates for other Microsoft products" in Settings. So I turned it on, and resumed Windows Updates which was paused. And, it got update for WSL to 5.10.60.1 (the version was 5.10.16), and it got a installation of WSLg too... Everything seems work properly now.

So, my question for now is, why does not it install the newest version of WSL, and install WSLg too when I enabled it? Is it a mistake, or designed to be like that?

Windows Build: 22478

**Comment by onomatopellan** [comment]
There is a new way of installing WSL2 from the Microsoft Store. It contains the last kernel and WSLg versions and will be updated more frequently.

https://devblogs.microsoft.com/commandline/a-preview-of-wsl-in-the-microsoft-store-is-now-available/

**Reply by IcyCod** [reply]
Thanks!
